src/scalib_ext/ranklib/src/histogram.rs

```diff
@@ -31,54 +39,143 @@ impl Histogram for F64Hist {
             state: vec![0.0; size],
             fft: planner.plan_fft_forward(2 * size),
             ifft: planner.plan_fft_inverse(2 * size),
+            state_upper: vec![0.0; size],
+            scale: 1.0,
         }
     }
+    fn type_str(&self) -> &'static str {
+        return "F64Hist";
+    }
+    fn rescale(&self) -> Self {
```

Comment:
please  document the functions with some descriptive comments to better explain what you're doing. See rlda.rs for an example